gui/src/renderer/components/Modal.tsx

```diff
@@ -98,6 +106,16 @@ interface IModalAlertProps {
 
 export class ModalAlert extends Component<IModalAlertProps> {
   public render() {
+    const modalContainer = document.getElementById(MODAL_CONTAINER_ID);
+    if (modalContainer !== null) {
+      return ReactDOM.createPortal(this.renderModal(), modalContainer);
+    } else {
+      log.error('Modal container not found when rendering modal');
```

Comment:
You can `throw Error("Modal container not found when rendering modal")`, we usually catch those in the error boundary and log them automatically. Also, this is likely a critical error which should not happen.

Reply:
Okay :+1: Should critical errors be handled differently? Should I skip the `modalContainer !== null` check?